File: dart/scrabble-score/lib/scrabble_score.dart
final String scoreMap = '|AEIOULNRST|DG|BCMP|FHVWY|K|||JX||QZ';

int score(String word) => word.toUpperCase().split('').fold(
    0,
    (sum, letter) =>
        sum +
        scoreMap
            .split('')
            .takeWhile((l) => l != letter)
            .where((l) => l == '|')
            .length);

File: kotlin/scrabble-score/src/main/kotlin/ScrabbleScore.kt
object ScrabbleScore {
    private const val scoreMap: String = "|AEIOULNRST|DG|BCMP|FHVWY|K|||JX||QZ"

    private fun scoreLetter(c: Char): Int = scoreMap.takeWhile { it != c }.filter { it == '|' }.count()

    fun scoreWord(word: String): Int = word.toUpperCase().map { scoreLetter(it) }.sum()
}
